### PR TITLE
test: each test case for browser

### LINKED ISSUE #3085: test.each does not work in browser mode
### Describe the bug

Trying to run a `test.each` or `it.each` test in browser mode fails with the error "import_util.default.format is not a function"

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-gehlnr?file=test/basic.test.ts

### System Info

```shell
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i5-10210U CPU @ 1.60GHz
    Memory: 6.97 GB / 15.78 GB
  Binaries:
    Node: 16.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files\nodejs\yarn.CMD
    npm: 8.15.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (111.0.1661.54)
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    @vitest/browser: ^0.29.7 => 0.29.7
    @vitest/ui: latest => 0.29.7
    vite: latest => 4.2.1
    vitest: latest => 0.29.7
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### LINKED ISSUE #3124: Browser mode hanging on blank white screen, failing to load vitest/utils
### Describe the bug

When trying to run a test in browser mode (playwright) with a basic example next.js application, the browser window opens but hangs on a blank screen and fails to load vitest/utils in the background.

Screenshot 2023-04-03 at 21 17 38

### Reproduction

Reproduction repository, based on vitest next.js example: https://github.com/ludwigbacklund/vitest-browser-bug-repro

Just run `yarn test`, it will open a browser window and hang on a white screen. Check the console/network log.

### System Info

```shell
System:
    OS: macOS 13.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 774.77 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nodenv/versions/16.13.1/bin/node
    Yarn: 1.22.15 - ~/.nodenv/versions/16.13.1/bin/yarn
    npm: 8.1.2 - ~/.nodenv/versions/16.13.1/bin/npm
  Browsers:
    Chrome: 111.0.5563.146
    Safari: 16.4
  npmPackages:
    @vitejs/plugin-react: latest => 3.1.0 
    @vitest/browser: ^0.29.8 => 0.29.8 
    vitest: latest => 0.29.8
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by Aslemammad**
This tries to resolve issue #3085 since it cannot be reproduced in linux and it works as expected, but adding a test case for more safety! 
Resolves #3124

### CODE REVIEW

**Comment by Aslemammad** [comment]
We need to add a browser + windows ci!

File: test/browser/package.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
     "test": "pnpm run test:webdriverio && pnpm run test:playwright",
-    "test:webdriverio": "PROVIDER=webdriverio node --test specs/",
-    "test:playwright": "PROVIDER=playwright node --test specs/",
+    "test:webdriverio": "cross-env PROVIDER=webdriverio node --test specs/",
+    "test:playwright": "cross-env PROVIDER=playwright node --test specs/",
     "coverage": "vitest --coverage.enabled --coverage.provider=istanbul"
   },
   "devDependencies": {
     "@vitest/browser": "workspace:*",
+    "cross-env": "^7.0.3",
```
Comment (sheremet-va) [inline test/browser/package.json:13]: We don't need `cross-env`, this functionality is built into pnpm

**Review by sheremet-va — CHANGES_REQUESTED** [review]

File: test/browser/package.json (hunk 1 above)
Comment (Aslemammad) [reply test/browser/package.json:13]: The thing is that the ci failed before in the previous commits! Any thoughts?

```

> @vitest/monorepo@0.29.8 test:browser:webdriverio D:\a\vitest\vitest
> npm -C test/browser run test:webdriverio


> test:webdriverio
> PROVIDER=webdriverio node --test specs/

'PROVIDER' is not recognized as an internal or external command,
operable program or batch file.
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.

```

Comment (Aslemammad) [reply test/browser/package.json:13]: let me try pnpm instead of npm

**Comment by Aslemammad** [comment]
The solution for #3124 was that vite didn't allow accessing the dist/utils.js file because of `config.fs.strict`, so I made it false so the files can be accessed.

**Comment by Aslemammad** [comment]
For the `each` case, I tested it in different platforms and it couldn't be reproduced, same with the tests.

**Comment by veaba** [comment]
Not yet fixed on windows.

image

**Comment by sheremet-va** [comment]
> Not yet fixed on windows.
> 
> image

This is why the issue is not closed.

**Comment by veaba** [comment]
Sorry, I see that this PR has been merged.

I just clone this repo and use `pnpm link` to import vitest into my project.

**Pull request merged**